src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -203,31 +246,88 @@ std::string MutationReplaceBinaryOperator::GetFunctionName(
+                                                       llvm::APSInt::get(1))) {
+      result += "_rhs_one";
+    } else if (rhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                       llvm::APSInt::get(-1))) {
+      result += "_rhs_minus_one";
+    }
+
+    bool lhs_is_int =
+        binary_operator_.getLHS()->EvaluateAsInt(eval_result, ast_context);
+    if (lhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                llvm::APSInt::get(0))) {
+      result += "_lhs_zero";
+    } else if (lhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                       llvm::APSInt::get(1))) {
+      result += "_lhs_one";
+    } else if (lhs_is_int && llvm::APSInt::isSameValue(eval_result.Val.getInt(),
+                                                       llvm::APSInt::get(-1))) {
+      result += "_lhs_minus_one";
+    }
+  }
+
   return result;
 }
 
 void MutationReplaceBinaryOperator::GenerateArgumentReplacement(
     const std::string& arg1_evaluated, const std::string& arg2_evaluated,
-    std::stringstream& new_function, int& mutant_offset) const {
+    clang::ASTContext& ast_context, std::stringstream& new_function,
+    int& mutant_offset) const {
   if (!binary_operator_.isAssignmentOp()) {
+    clang::Expr::EvalResult lhs_eval_result;
+    clang::Expr::EvalResult rhs_eval_result;
+    bool rhs_is_int =
+        binary_operator_.getRHS()->EvaluateAsInt(rhs_eval_result, ast_context);
+    bool lhs_is_int =
+        binary_operator_.getLHS()->EvaluateAsInt(lhs_eval_result, ast_context);
+
     // LHS
-    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
-                 << mutant_offset << ")) return " << arg1_evaluated << ";\n";
-    mutant_offset++;
+    // These cases are equivalent to constant replacement with the respective
+    // constants
+    if (!(lhs_is_int && (llvm::APSInt::isSameValue(lhs_eval_result.Val.getInt(),
+                                                   llvm::APSInt::get(0)) ||
+                         llvm::APSInt::isSameValue(lhs_eval_result.Val.getInt(),
+                                                   llvm::APSInt::get(1)) ||
+                         llvm::APSInt::isSameValue(lhs_eval_result.Val.getInt(),
+                                                   llvm::APSInt::get(-1))))) {
+      new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
+                   << mutant_offset << ")) return " << arg1_evaluated << ";\n";
+      mutant_offset++;
+    }
 
     // RHS
-    new_function << "  if (__dredd_enabled_mutation(local_mutation_id + "
-                 << mutant_offset << ")) return " << arg2_evaluated << ";\n";
-    mutant_offset++;
+    // These cases are equivalent to constant replacement with the respective
+    // constants
```

Comment:
And I'd move the declaration of RHS stuff to after this comment.

Reply:
Done.